### PR TITLE
feat: Added new recommendation algorithm to server

### PR DESCRIPTION
**Pull request by arjunr00**
* Now requires user preferences and book list
* Currently recommends a single book (can update in a later PR to return books whose scores are above a given threshold)
* Input requests must be POST requests with a body of the form (expressed as a TypeScript interface):
```typescript
interface Book {
  books: {
    isbn?: string;
    title: string;
    author?: string;
    year?: number;
    publisher?: string;
}
```

### CODE REVIEW

File: RecAlgo/test2.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from utils import Book
+import pandas as pd
+import numpy as np
+
+
+# In[2]:
+
+
+'''
+test_book: a Book instance, contains six attributes: [ISBN, title, author, year, publisher, imageUrl].
```
Comment (BryanPan342) [inline RecAlgo/test2.py]: whats this file about?

**Review by BryanPan342 — COMMENTED** [review]
had a comment on a single file

File: RecAlgo/test2.py (hunk 1 above)
Comment (arjunr00) [reply RecAlgo/test2.py]: oh oops i shouldn't have committed this my bad, i converted the test notebook to a .py file so it was easier to run

**Review by BryanPan342 — APPROVED** [review]
LGTM

**Pull request merged**